src/backend/api/Fusion.Resources.Api/Notifications/RequestsNotificationHandler.cs

```diff
@@ -0,0 +1,263 @@
+            var contract = await orgResolver.ResolveContractAsync(request.Project.OrgProjectId, request.Contract.OrgContractId);
+
+            if (contract == null)
+                throw new InvalidOperationException($"Cannot resolve contract for request {request.Id}");
+
+            var externalCompanyRep = contract.ExternalCompanyRep.GetActiveInstance();
+            var externalContractRep = contract.ExternalContractRep.GetActiveInstance();
+            var delegates = await mediator.Send(GetContractDelegatedRoles.ForContract(request.Project.OrgProjectId, contract.Id));
+            var externalDelegates = delegates.Where(o => o.Classification == DbDelegatedRoleClassification.External).ToList();
+            var recipients = new List<Guid>();
+
+            if (externalCompanyRep?.AssignedPerson?.AzureUniqueId != null && !recipients.Contains(externalCompanyRep.AssignedPerson.AzureUniqueId.Value))
+                recipients.Add(externalCompanyRep.AssignedPerson.AzureUniqueId.Value);
+
+            if (externalContractRep?.AssignedPerson?.AzureUniqueId != null && !recipients.Contains(externalContractRep.AssignedPerson.AzureUniqueId.Value))
+                recipients.Add(externalContractRep.AssignedPerson.AzureUniqueId.Value);
+
+            var distinctDelegates = externalDelegates.Where(d => !recipients.Contains(d.Person.AzureUniqueId))
+                .Select(d => d.Person.AzureUniqueId)
+                .Distinct();
+
+            recipients.AddRange(distinctDelegates);
+
+            return recipients;
+        }
+
+        private async Task<List<Guid>> CalculateInternalCRRecipientsAsync(QueryPersonnelRequest request)
+        {
+            var contract = await ResolveContractAsync(request);
+            var companyRep = contract.CompanyRep.GetActiveInstance();
+            var contractRep = contract.ContractRep.GetActiveInstance();
+            var delegates = await mediator.Send(GetContractDelegatedRoles.ForContract(request.Project.OrgProjectId, contract.Id));
+            var internalDelegates = delegates.Where(o => o.Classification == DbDelegatedRoleClassification.Internal).ToList();
+
+            var recipients = new List<Guid>();
+
+            if (companyRep?.AssignedPerson?.AzureUniqueId != null && !recipients.Contains(companyRep.AssignedPerson.AzureUniqueId.Value))
+                recipients.Add(companyRep.AssignedPerson.AzureUniqueId.Value);
+
+            if (contractRep?.AssignedPerson?.AzureUniqueId != null && !recipients.Contains(contractRep.AssignedPerson.AzureUniqueId.Value))
+                recipients.Add(contractRep.AssignedPerson.AzureUniqueId.Value);
+
+            var distinctDelegates = internalDelegates.Where(d => !recipients.Contains(d.Person.AzureUniqueId))
+                .Select(d => d.Person.AzureUniqueId)
+                .Distinct();
+
+            recipients.AddRange(distinctDelegates);
+
+            return recipients;
+        }
+
+        private class NotificationDescription
+        {
```

Comment:
Like mentioned before.
Incomplete paragraph's & listitems should be complete or ignored...

Also the image thingy,... ? or...

Reply:
Discussed paragraphs above.

Image thingy will require total re-write to use Cards, wouldn't it?

Reply:
Postponing this.